.github/actions/pr-comment/action.yml

```diff
@@ -0,0 +1,67 @@
+name: PR Comment
+description: Action that comments on a given PR
+inputs:
+  comment:
+    type: string
+    required: true
+    description: The body of the comment, in GitHub Markdown format
+  pr:
+    type: number
+    required: false
+    description: The Pull Request Number that will be commented on. Defaults to the PR associated with the callers event
+  edit-last:
+    type: boolean
+    required: false
+    default: false
+    description: Edit the last comment by 'github-actions[bot]' instead of creating a new one
+  token:
+    type: string
+    required: false
+    description: A GitHub Token/PAT that allows commenting on the given PR. Defaults to github.token
+outputs:
+  comment-link:
+    description: A link to the added comment
+    value: ${{ steps.comment.outputs.link }}
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set Non-Required Inputs
+      shell: bash
+      run: |
+        # Set PR
+        if [[ -z "${{ inputs.pr }}" ]]; then
+          if [[ -n "${{ github.event.issue }}" ]]; then
+            echo "PR=${{ github.event.issue.number }}" >> $GITHUB_ENV
+          elif [[ -n "${{ github.event.pull_request }}" ]]; then
+            echo "PR=${{ github.event.pull_request.number }}" >> $GITHUB_ENV
+          else
+            echo "::error::Couldn't infer PR number from event payload, please enter as an input"
+            exit 1
+          fi
+        else
+          echo "PR=${{ inputs.pr }}" >> $GITHUB_ENV
```

Comment:
Does this need any checking that the PR exists? That it is open?

Reply:
I don't think it needs to check that it is open, because commenting on closed PRs (even though it might be a little strange) is still perfectly valid. Maybe there is some use case where we would want to do that. 
As for the case where it doesn't exist yet, I'm happy for the call to `gh pr comment` to give a sufficient error, as they already do:
```
$ gh pr comment 9999 --body 'blah'
GraphQL: Could not resolve to a PullRequest with the number of 9999. (repository.pullRequest)
$ echo $?
1
```